libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -1490,51 +1479,44 @@ tiledb_datatype_t ArrowAdapter::to_tiledb_format(
     }
 }
 
-// FIXME: Add more types, maybe make it a map
 enum ArrowType ArrowAdapter::to_nanoarrow_type(std::string_view sv) {
-    if (sv == "i")
-        return NANOARROW_TYPE_INT32;
-    else if (sv == "c")
-        return NANOARROW_TYPE_INT8;
-    else if (sv == "C")
-        return NANOARROW_TYPE_UINT8;
-    else if (sv == "s")
-        return NANOARROW_TYPE_INT16;
-    else if (sv == "S")
-        return NANOARROW_TYPE_UINT16;
-    else if (sv == "I")
-        return NANOARROW_TYPE_UINT32;
-    else if (sv == "l")
-        return NANOARROW_TYPE_INT64;
-    else if (sv == "L")
-        return NANOARROW_TYPE_UINT64;
-    else if (sv == "f")
-        return NANOARROW_TYPE_FLOAT;
-    else if (sv == "g")
-        return NANOARROW_TYPE_DOUBLE;
-    else if (sv == "u")
-        return NANOARROW_TYPE_STRING;
-    else if (sv == "U")
-        return NANOARROW_TYPE_LARGE_STRING;
-    else if (sv == "b")
-        return NANOARROW_TYPE_BOOL;
-    else if (sv == "tss:")
-        return NANOARROW_TYPE_INT64;  // NB time resolution set indepedently
-    else if (sv == "tsm:")
-        return NANOARROW_TYPE_INT64;  // NB time resolution set indepedently
-    else if (sv == "tsn:")
-        return NANOARROW_TYPE_INT64;  // NB time resolution set indepedently
-    else if (sv == "tsu:")
-        return NANOARROW_TYPE_INT64;  // NB time resolution set indepedently
-    else if (sv == "tdD")
-        return NANOARROW_TYPE_INT32;  // R Date: fractional days since epoch
-    else if (sv == "z")
-        return NANOARROW_TYPE_BINARY;
-    else if (sv == "Z")
-        return NANOARROW_TYPE_LARGE_BINARY;
-    else
-        throw TileDBSOMAError(
-            fmt::format("ArrowAdapter: Unsupported Arrow format: {} ", sv));
+    std::map<std::string_view, enum ArrowType> _to_nanoarrow_type_map = {
+        {"i", NANOARROW_TYPE_INT32},        {"c", NANOARROW_TYPE_INT8},
+        {"C", NANOARROW_TYPE_UINT8},        {"s", NANOARROW_TYPE_INT16},
+        {"S", NANOARROW_TYPE_UINT16},       {"I", NANOARROW_TYPE_UINT32},
+        {"l", NANOARROW_TYPE_INT64},        {"L", NANOARROW_TYPE_UINT64},
+        {"f", NANOARROW_TYPE_FLOAT},        {"g", NANOARROW_TYPE_DOUBLE},
+        {"u", NANOARROW_TYPE_STRING},       {"U", NANOARROW_TYPE_LARGE_STRING},
+        {"b", NANOARROW_TYPE_BOOL},         {"tss:", NANOARROW_TYPE_TIMESTAMP},
+        {"tsm:", NANOARROW_TYPE_TIMESTAMP}, {"tsn:", NANOARROW_TYPE_TIMESTAMP},
+        {"tsu:", NANOARROW_TYPE_TIMESTAMP}, {"tdD", NANOARROW_TYPE_TIMESTAMP},
+        {"z", NANOARROW_TYPE_BINARY},       {"Z", NANOARROW_TYPE_LARGE_BINARY},
+    };
+
+    try {
+        return _to_nanoarrow_type_map.at(sv);
+    } catch (const std::out_of_range& e) {
+        throw std::out_of_range(
+            fmt::format("ArrowAdapter: Unsupported nanoarrow format: {} ", sv));
```

Comment:
I'm not sure end-users will know what `nanoarrow` is. Suggest reverting to the original us of `Arrow`.

Also, suggest using the term "type", not "format".  E.g.,

> ArrowAdaptor: Unsupported Arrow type: ..."

Reply:
Also, this is going to throw an error using the internal type code (e.g., `ttu`), rather than the user-friendly `Time64('us')`.  Can we improve the error message in some way?  Or alternatively, help the user figure out what the internal type code means?

Best, worst idea I have is to include a pointer to the docs ([here](https://arrow.apache.org/docs/format/CDataInterface.html#data-type-description-format-strings)), but it would be better if the error message just said: 
> ArrowAdaptor: Unsupported Arrow type: Time64('us')